unit_tests/test_matching_algorithms/test_preferred_partners_matching_algorithm.py

```diff
@@ -0,0 +1,126 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import uuid
+from typing import Optional
+
+from d3a_interface.data_classes import Offer, BidOfferMatch, Bid
+from pendulum import DateTime
+
+from myco_api_client.matching_algorithms.preferred_partners_algorithm import (
+    PreferredPartnersMatchingAlgorithm)
+
+
+class TestPreferredPartnersMatchingAlgorithm:
+
+    @staticmethod
+    def offer_factory(additional_data: dict = None):
+        additional_data = additional_data or {}
+        return Offer(
+            **{"id": str(uuid.uuid4()),
+               "creation_time": DateTime.now(),
+               "time_slot": DateTime.now(),
+               "price": 10,
+               "energy": 30,
+               "seller": "seller",
+               "seller_id": "seller_id",
+               "seller_origin": "seller",
+               "seller_origin_id": "seller_id",
+               **additional_data})
+
+    @staticmethod
+    def bid_factory(additional_data: Optional[dict] = None):
+        additional_data = additional_data or {}
+        return Bid(
+            **{"id": str(uuid.uuid4()),
+               "creation_time": DateTime.now(),
+               "time_slot": DateTime.now(),
+               "price": 10,
+               "energy": 30,
+               "buyer": "buyer",
+               "buyer_id": "buyer_id",
+               "buyer_origin": "buyer",
+               "buyer_origin_id": "buyer_id",
+               **additional_data})
+
+    def test_perform_trading_partners_matching(self):
+        offer = self.offer_factory().serializable_dict()
+        bid = self.bid_factory(
+            {"requirements": [{"trading_partners": [offer["seller_id"]]}]}
+        ).serializable_dict()
+        assert PreferredPartnersMatchingAlgorithm._perform_trading_partners_matching(
+            market_id="market", time_slot="2021-10-06T12:00",
+            bids=[bid], offers=[offer]) == [
+                   BidOfferMatch(bids=[bid], offers=[offer],
+                                 market_id="market",
+                                 trade_rate=bid["energy_rate"],
+                                 selected_energy=30,
+                                 time_slot="2021-10-06T12:00").serializable_dict()]
```

Comment:
Nit: indentation is off here

Reply:
The number of spaces is not a multiple of 4, and the indentation is double of what is needed. One possible correct indentation would be this:
```suggestion
        assert PreferredPartnersMatchingAlgorithm._perform_trading_partners_matching(
            market_id="market", time_slot="2021-10-06T12:00",
            bids=[bid], offers=[offer]) == [
                BidOfferMatch(bids=[bid], offers=[offer],
                              market_id="market",
                              trade_rate=bid["energy_rate"],
                              selected_energy=30,
                              time_slot="2021-10-06T12:00").serializable_dict()]
```